### PR TITLE
[LoongArch][Codegen] Add support for TLSDESC

### PR DESCRIPTION
**Pull request by wangleiat**
The implementation only enables when the `-enable-tlsdesc` option is
passed and the TLS model is `dynamic`.

LoongArch's GCC has the same option(-mtls-dialet=) as RISC-V.

### CODE REVIEW

**Comment by wangleiat** [comment]
@xen0n @xry111

**Comment by xry111** [comment]
> [wangleiat](https://github.com/wangleiat) wants to merge 1 commit into [users/wangleiat/spr/main.loongarchcodegen-add-support-for-tlsdesc-1](https://github.com/llvm/llvm-project/tree/users/wangleiat/spr/main.loongarchcodegen-add-support-for-tlsdesc-1) from [users/wangleiat/spr/loongarchcodegen-add-support-for-tlsdesc-1](https://github.com/llvm/llvm-project/tree/users/wangleiat/spr/loongarchcodegen-add-support-for-tlsdesc-1)

Hmm, it looks like the target branch is wrong?

**Comment by xry111** [comment]
> > > [wangleiat](https://github.com/wangleiat) wants to merge 1 commit into [users/wangleiat/spr/main.loongarchcodegen-add-support-for-tlsdesc-1](https://github.com/llvm/llvm-project/tree/users/wangleiat/spr/main.loongarchcodegen-add-support-for-tlsdesc-1) from [users/wangleiat/spr/loongarchcodegen-add-support-for-tlsdesc-1](https://github.com/llvm/llvm-project/tree/users/wangleiat/spr/loongarchcodegen-add-support-for-tlsdesc-1)
> > 
> > 
> > Hmm, it looks like the target branch is wrong?
> 
> I used the [SPR](https://getcord.github.io/spr/) for stack-style submissions for the first time, and according to the [documentation](https://getcord.github.io/spr/user/stack.html#cherry-picking), it seems to be correct.

Ok, I didn't know SPR.  Stupid I...

File: clang/lib/Driver/ToolChains/CommonArgs.cpp [hunk 1]
```diff
@@ -737,7 +737,7 @@ bool tools::isTLSDESCEnabled(const ToolChain &TC,
   StringRef V = A->getValue();
   bool SupportedArgument = false, EnableTLSDESC = false;
   bool Unsupported = !Triple.isOSBinFormatELF();
-  if (Triple.isRISCV()) {
+  if (Triple.isRISCV() || Triple.isLoongArch()) {
```
Comment (SixWeining) [inline clang/lib/Driver/ToolChains/CommonArgs.cpp:740]: Is it better to sort alphabetically?

**Review by SixWeining — APPROVED** [review]
LGTM except a nit.

File: clang/lib/Driver/ToolChains/CommonArgs.cpp (hunk 1 above)
Comment (wangleiat) [reply clang/lib/Driver/ToolChains/CommonArgs.cpp:740]: Thanks, I will make the modifications.

**Comment by MaskRay** [comment]
Add a test to clang/test/Driver/tls-dialect.c

**Review by MaskRay — APPROVED** [review]

**Pull request merged**